tests/abi/testabi_ks.c

```diff
@@ -221,8 +221,10 @@ static void test_ksReference (void)
 	succeed_if (ksAtCursor (ks1, 0) == k2, "head in dup wrong");
 	succeed_if (ksAtCursor (ks1, ksGetSize (ks1) - 1) == k1, "tail in dup wrong");
 
-	succeed_if (keyGetRef (k1) == 2, "reference counter after duplication of keyset");
-	succeed_if (keyGetRef (k2) == 2, "reference counter after ksdup");
+	// COW - key references stay the same
+	succeed_if (keyGetRef (k1) == 1, "reference counter after duplication of keyset");
+	succeed_if (keyGetRef (k2) == 1, "reference counter after ksdup");
```

Comment:
You should add tests for when the references change, i.e. after `ks1` has been modified.